mobile/src/main/java/org/openhab/habdroid/ui/OpenHABPreferencesActivity.java

```diff
@@ -323,70 +387,140 @@ private void onNoDefaultSitemap(Preference pref) {
+                @StringRes int resId = TextUtils.isEmpty(value) ? R.string.info_not_set
+                        : isWeakPassword(value) ? R.string.settings_openhab_password_summary_weak
+                        : R.string.settings_openhab_password_summary_strong;
+                return getString(resId);
+            });
+
+            updateIconColors(getPreferenceString(urlPrefKey, ""),
+                    getPreferenceString(userNamePrefKey, ""),
+                    getPreferenceString(passwordPrefKey, ""));
+        }
+
+        private Preference initEditorPreference(String key, PrefSummaryGenerator summaryGenerator) {
+            Preference preference = getPreferenceScreen().findPreference(key);
+            preference.setOnPreferenceChangeListener((pref, newValue) -> {
+                updateIconColors(getActualValue(pref, newValue, mUrlPreference),
+                        getActualValue(pref, newValue, mUserNamePreference),
+                        getActualValue(pref, newValue, mPasswordPreference));
+                pref.setSummary(summaryGenerator.getSummary((String) newValue));
+                return true;
             });
-            updateTextPreferenceSummary(pref, summaryFormatResId, null, isPassword);
+            preference.setSummary(summaryGenerator.getSummary(getPreferenceString(key, "")));
+            return preference;
+        }
+
+        private String getActualValue(Preference pref, Object newValue, Preference reference) {
+            return pref == reference ? (String) newValue : getPreferenceString(reference, "");
+        }
+
+        private void updateIconColors(String url, String userName, String password) {
+            updateIconColor(mUrlPreference, () -> {
+                if (!TextUtils.isEmpty(url)) {
+                    return isConnectionHttps(url) ? R.color.pref_icon_green : R.color.pref_icon_red;
+                }
+                return null;
+            });
+            updateIconColor(mUserNamePreference, () -> {
+                if (!TextUtils.isEmpty(url)) {
+                    return TextUtils.isEmpty(userName) ? R.color.pref_icon_red : R.color.pref_icon_green;
+                }
+                return null;
+            });
+            updateIconColor(mPasswordPreference, () -> {
+                if (!TextUtils.isEmpty(url)) {
+                    if (TextUtils.isEmpty(password)) {
+                        return R.color.pref_icon_red;
+                    } else if (isWeakPassword(password)) {
+                        return R.color.pref_icon_orange;
+                    } else {
+                        return R.color.pref_icon_green;
+                    }
+                }
+                return null;
+            });
+        }
+
+        private void updateIconColor(Preference pref, IconColorGenerator colorGenerator) {
+            Drawable icon = pref.getIcon();
+            if (icon == null) {
+                return;
```

Comment:
The icon will be null here when the subpage will be opend, at least on api 22. Instead of getting the icon via `pref.getIcon()` there could be a mapping of `Preference` to icon name.

Reply:
I'll look at the Preference.getIcon implementation again.

Reply:
OK, this behavior indeed was changed in Android 7.1: https://github.com/LineageOS/android_frameworks_base/commit/f049e24c644912a308ca283ba1c75d678a09f4a8
I'll hardcode the resource IDs (make them a parameter of initEditorPreference).

Reply:
Fix: https://github.com/maniac103/openhab.android/commit/dddfc22070cb11873b5fa4181a20860a950417ce ... tested down to API 16.